### PR TITLE
Add MaterialData for Stone

### PR DESCRIPTION
**Pull request by jimmikaelkael**
This allow to identify the seven type of stone blocks: normal, granite, polished granite, diorite, polished diorite, andesite and polished andesite.

---

_Edited by turt2live_

**Related links:**
- Glowstone: None needed

### CODE REVIEW

File: src/main/java/org/bukkit/StoneType.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+import java.util.Map;
+
+import com.google.common.collect.Maps;
+
+/**
+ * Represents the seven different types of stone.
+ */
+public enum StoneType {
+
```
Comment (turt2live) [inline src/main/java/org/bukkit/StoneType.java]: Extra whitespace

File: src/main/java/org/bukkit/StoneType.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    private final byte data;
+    private final static Map<Byte, StoneType> BY_DATA = Maps.newHashMap();
+
+    private StoneType(final int data) {
+        this.data = (byte) data;
+    }
+
+    /**
+     * Gets the associated data value representing this type
```
Comment (turt2live) [inline src/main/java/org/bukkit/StoneType.java]: Missing period

File: src/main/java/org/bukkit/StoneType.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+     *
+     * @return A byte containing the data value of this stone type
```
Comment (turt2live) [inline src/main/java/org/bukkit/StoneType.java:50]: ~~`@return a byte...` (case)~~

I'm dumb

File: src/main/java/org/bukkit/StoneType.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+     * @deprecated Magic value
+     */
+    @Deprecated
+    public byte getData() {
+        return data;
+    }
+
+    /**
+     * Gets the StoneType with the given data value
```
Comment (turt2live) [inline src/main/java/org/bukkit/StoneType.java]: Missing period

File: src/main/java/org/bukkit/StoneType.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+     *
+     * @param data Data value to fetch
```
Comment (turt2live) [inline src/main/java/org/bukkit/StoneType.java]: `@param data The data value to fetch`

File: src/main/java/org/bukkit/material/Stone.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
+package org.bukkit.material;
+
+import org.bukkit.StoneType;
+import org.bukkit.Material;
+
+/**
+ * Represents the different types of stone.
+ */
+public class Stone extends MaterialData {
```
Comment (turt2live) [inline src/main/java/org/bukkit/material/Stone.java:9]: Missing blank line after declaration

File: src/main/java/org/bukkit/material/Stone.java [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+        super(Material.STONE);
+    }
+
+    public Stone(StoneType stone) {
+        this();
+        setType(stone);
+    }
+
+    /**
+     *
```
Comment (turt2live) [inline src/main/java/org/bukkit/material/Stone.java]: Remove excess line

File: src/main/java/org/bukkit/material/Stone.java [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+    public Stone(final int type) {
+        super(type);
+    }
+
+    public Stone(final Material type) {
+        super(type);
+    }
+
+    /**
+     *
```
Comment (turt2live) [inline src/main/java/org/bukkit/material/Stone.java]: https://github.com/GlowstoneMC/Glowkit/pull/30/files#r20180814

File: src/main/java/org/bukkit/material/Stone.java [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+     * @deprecated Magic value
+     */
+    @Deprecated
+    public Stone(final int type, final byte data) {
+        super(type, data);
+    }
+
+    /**
+     *
```
Comment (turt2live) [inline src/main/java/org/bukkit/material/Stone.java]: https://github.com/GlowstoneMC/Glowkit/pull/30/files#r20180814

File: src/main/java/org/bukkit/material/Stone.java [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+     * @deprecated Magic value
+     */
+    @Deprecated
+    public Stone(final Material type, final byte data) {
+        super(type, data);
+    }
+
+    /**
+     * Gets the current type of this stone
```
Comment (turt2live) [inline src/main/java/org/bukkit/material/Stone.java]: Missing period

File: src/main/java/org/bukkit/material/Stone.java [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+     *
+     * @return StoneType of this stone
+     */
+    public StoneType getType() {
+        return StoneType.getByData(getData());
+    }
+
+    /**
+     * Sets the type of this stone
```
Comment (turt2live) [inline src/main/java/org/bukkit/material/Stone.java]: Missing period

**Comment by jimmikaelkael** [comment]
@turt2live Rebased on top of master and applied the requested format fixes.

**Comment by SpaceManiac** [comment]
Merged in 267a795149a14250209b822ec99804c335150a56.

**Pull request closed without merging**